DataManagementSystem/DB/FileCatalogComponents/DirectoryMetadata.py

```diff
@@ -350,6 +384,17 @@ def getDirectoryMetadata(self, path, credDict, inherited=True, owndata=True):
       for meta in result['Value']:
         metaOwnerDict[meta] = 'OwnParameter'
 
+    if strip_suffix:
+      suffix = self._getMetaNameSuffix(credDict)
+      metaDict = {key.replace(suffix, '') if key.endswith(suffix)
+                  else key: value for key, value in metaDict.iteritems()}
+
+      metaOwnerDict = {key.replace(suffix, '') if key.endswith(suffix)
+                       else key: value for key, value in metaOwnerDict.iteritems()}
+
+      metaTypeDict = {key.replace(suffix, '') if key.endswith(suffix)
+                      else key: value for key, value in metaTypeDict.iteritems()}
```

Comment:
What about
```
def stripSuffix(aDict, suffix):
  return {key.replace(suffix, '') if key.endswith(suffix)
              else key: value for key, value in aDict.iteritems()}

if strip_suffix:
  metaDict = stripSuffix(metaDict, suffix)
  metaOwnerDict = stripSuffix(metaOwnerDict, suffix)
  metaTypeDict = stripSuffix(metaTypeDict, suffix)

```

Reply:
Not there anymore. Added mixing classes as suggested by Andre.